packages/medusa/src/api/routes/admin/discounts/create-discount.ts

```diff
@@ -154,9 +157,31 @@ export class AdminPostDiscountsDiscountRule {
   @IsNotEmpty()
   allocation: string
 
-  // @IsOptional()
-  // @IsArray()
-  // @ValidateNested({ each: true })
-  // @Type(() => Condition)
-  // conditions?: Condition[]
+  @IsOptional()
+  @IsArray()
+  @ValidateNested({ each: true })
+  @Type(() => AdminCreateCondition)
+  conditions?: AdminCreateCondition[]
+}
+
+export class AdminCreateCondition {
+  @IsString()
+  operator: DiscountConditionOperator
+
+  @IsString()
+  resource_type: DiscountConditionType
+
+  @IsArray()
+  @IsString({ each: true })
+  resource_ids: string[]
```

Comment:
Not sure what you think but maybe we should change this to be `products, collections, types, tags` and then infer the resource type based on the populated field?

My reasoning is just that we generally have equal structures in the create/update payload and the responses, so it might be confusing if that isn't consistent throughout the API.

Reply:
You raise a good point. The approach here is a result of the implementation in the service layer in which I've abstracted away the specific resource type when adding conditions. Though, I see how it make sense from an API perspective since these are exposed to the user. And would also be consistent with our Tax API.

Will change! 👍